pkg/state/types.go

```diff
@@ -1842,3 +1842,50 @@ func (d *DegraphqlRoute) EqualWithOpts(d2 *DegraphqlRoute, ignoreID bool) bool {
 
 	return reflect.DeepEqual(d1Copy, d2Copy)
 }
+
+// Partial represents a partial in Kong.
+// It adds some helper methods along with Meta to the original Partial object.
+type Partial struct {
+	kong.Partial `yaml:",inline"`
+	Meta
+}
+
+// Identifier returns the Partial name or ID.
+func (p *Partial) Identifier() string {
+	if p.Name != nil {
+		return *p.Name
+	}
+	return *p.ID
+}
+
+// Console returns an entity's identity in a human
+// readable string.
+func (p *Partial) Console() string {
+	return p.FriendlyName()
+}
+
+// Equal returns true if partials p and p2 are equal.
+func (p *Partial) Equal(p2 *Partial) bool {
+	return p.EqualWithOpts(p2, false, false)
+}
+
+// EqualWithOpts returns true if partials p and p2 are equal.
+// If ignoreID is set to true, IDs will be ignored while comparison.
+// If ignoreTS is set to true, timestamp fields will be ignored.
+func (p *Partial) EqualWithOpts(p2 *Partial, ignoreID, ignoreTS bool) bool {
```

Comment:
This can be a bit hard to reason about at call sites having 2 bool arguments.

WDYT about making this access functional options and providing e.g. `IgnoreID()` and `IgnoreTS()` option funcs which would act on `*Partial` (clearing `.ID` and timestamps respectively)? This would not only make this clearer to reason about at call site but also possible to be used with different sets of these options (e.g. ignoring only timestamps).

Reply:
I agree but I would like to keep it this way, so that it adheres to the formats other entities use for the same operation. If we need to change it, we should do it in all places to make things uniform. This can be tracked as a separate improvement.

Reply:
Fair enough, do you mind creating an issue for this?

Reply:
Done.
https://github.com/Kong/go-database-reconciler/issues/223